addons/website_sale/models/website.py

```diff
@@ -270,11 +270,11 @@ def get_current_pricelist(self):
             if available_pricelists and pricelist not in available_pricelists:
                 # If there is at least one pricelist in the available pricelists
                 # and the chosen pricelist is not within them
-                # it then choose the first available pricelist.
+                # it then choose the first available pricelist based on _order of the model.
                 # This can only happen when the pricelist is the public user pricelist and this pricelist is not in the available pricelist for this localization
                 # If the user is signed in, and has a special pricelist (different than the public user pricelist),
                 # then this special pricelist is amongs these available pricelists, and therefore it won't fall in this case.
-                pricelist = available_pricelists[0]
+                pricelist = available_pricelists.sorted()[0]
```

Comment:
Might have been even better to handle it in `get_pricelist_available` (and even the cached method `_get_pl_partner_order`), but it might be safer to stick to this localized patch for now :ok_hand: